device/esp_tinyusb/tinyusb.c

```diff
@@ -66,8 +78,18 @@ esp_err_t tinyusb_driver_install(const tinyusb_config_t *config)
     return ESP_OK;
 }
 
-esp_err_t tinyusb_driver_uninstall()
+esp_err_t tinyusb_driver_uninstall(void)
 {
+    esp_err_t ret = tusb_stop_task();
+
+    if (ret != ESP_OK) {
+        return ret;
+    }
+
+    if (!tusb_teardown()) {
+        return ESP_ERR_NOT_FINISHED;
+    }
+
```

Comment:
@roma-jam Should we restore the USB to the default function (USB-Serial-JTAG) at the end of uninstall?

Reply:
@leeebo I don't think that would be a good idea, as for user it is transparent in that case. 
Even, IDK, whether we have such feature before installing tinyUSB driver? 

Just in case: do we have such call (maybe in rom) to enable this feature? Or you thinking about configure the PHY during teardown in the way, to enable USB-Serial-JTAG?

Reply:
The USB-Serial-JTAG is a hardware circuit that can respond to the host's requests and is enabled by default. Users can use the USB-Serial-JTAG for USB download without any code in APP.

But after tinyUSB install then uninstalled, the USB phy not be switched back to USB-Serial-JTAG

Reply:
@roma-jam @leeebo as a user of the library in esp32s3 I think teardown/uninstall should not auto-reinstall jtag but there should be a convenience function that allows to restore it, if necessary it could be called automatically and be driven by Kconfig to provide best defaults. In my use case i want to be able to switch from usb cdc guest to msc usb host and viceversa and that is currently not possible without a device restart